File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/aspplus/samples/webforms/ctrlauth/templates/JS/RepeaterItemCollection.js
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

import System;
import System.Collections;
import System.Web;
import System.Web.UI;
import System.Web.Util;

package TemplateControlSamplesJS {

    public class RepeaterItemCollection implements ICollection {

        private var items:ArrayList;

        public function RepeaterItemCollection(items:ArrayList) {
            this.items = items;
        }

        public function get Count() : int {
            return items.Count;
        }

        public function get IsReadOnly() : Boolean {
            return false;
        }

        public function get IsSynchronized() : Boolean {
            return false;
        }

        public function get SyncRoot() : Object {
            return this;
        }

        public function Item(index:int) : RepeaterItem {
            return RepeaterItem(items[index]);
        }

        public function CopyTo(array:System.Array, index:int) : void {
            for (var e:IEnumerator = this.GetEnumerator(); e.MoveNext();)
                array.SetValue(e.Current, index++);
        }

        public function GetEnumerator() : IEnumerator {
            return items.GetEnumerator();
        }
    }
}

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/aspplus/samples/webforms/ctrlauth/templates/cs/RepeaterItemCollection.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System;
using System.Collections;
using System.Web;
using System.Web.UI;
using System.Web.Util;

namespace TemplateControlSamples {

    public class RepeaterItemCollection : ICollection {

        private ArrayList items;

        public RepeaterItemCollection(ArrayList items) {
            this.items = items;
        }
        
        public int Count {
            get {
                return items.Count;
            }
        }

        public bool IsReadOnly {
            get {
                return false;
            }
        }

        public bool IsSynchronized {
            get {
                return false;
            }
        }

        public object SyncRoot {
            get {
                return this;
            }
        }

        public RepeaterItem this[int index] {
            get {
                return (RepeaterItem)items[index];
            }
        }
        
        public void CopyTo(Array array, int index) {
            for (IEnumerator e = this.GetEnumerator(); e.MoveNext();)
                array.SetValue(e.Current, index++);
        }

        public IEnumerator GetEnumerator() {
            return items.GetEnumerator(); 
        }
    }
}